lib/bundler/ui/shell.rb

```diff
@@ -80,6 +81,9 @@ def silence
 
       # valimism
       def tell_me(msg, color = nil, newline = nil)
+        return if @told_messages.include? msg
```

Comment:
well in theory the same message can be intended twice in different context. Ideally, we'd limit de-duplication just to warning messages